woocommerce/woocommerce-ios
# [Min/Max Quantities] Add quantity rules row to product details

Part of #8960


## Description

We are adding read-only support for the [Min/Max Quantities](https://woocommerce.com/products/minmax-quantities/) extension in product details.

This PR adds a "Quantity Rules" row in the product details screen, when the product has min/max quantity rules set and the feature flag is enabled. The row details shows the first two quantity rules set for the product. For now, tapping the quantity rules row does nothing (navigation to a new view to be added in another PR).

### Changes

* Adds a new `readOnlyMinMaxQuantities` feature flag.
* Adds the min/max quantity settings to `ProductFormDataModel` and `EditableProductModel` (will be updated for product variations in a separate PR).
* Creates the row view model in `DefaultProductFormTableViewModel`.
* Adds the row to the settings section in `ProductFormActionsFactory` for all product types, always visible when the product has min/max quantity rules set.
* Adds unit tests.

## Testing instructions

### Prerequisites

1. Install and activate the [Min/Max Quantities](https://woocommerce.com/products/minmax-quantities/) extension.
2. Create or update at least one product to add quantity rules in the product details > General settings in wp-admin.

### To test

1. Build and run the app.
2. Go to the Products tab.
3. Select a product with quantity rules set, and confirm the Quantity Rules row appears with the expected details.
4. Select a product with no quantity rules set, and confirm the Quantity Rules row does not appear.

## Screenshots


/|Before|After
-|-|-
Product with min/max rules|![Simulator Screenshot - iPhone 14 Pro - 2023-04-27 at 16 55 13](https://user-images.githubusercontent.com/8658164/234918745-635573a2-ef74-4db5-a120-c8c8ffb1175a.png)|![Simulator Screenshot - iPhone 14 Pro - 2023-04-27 at 16 31 42](https://user-images.githubusercontent.com/8658164/234918814-6adf0b11-9203-4f3c-ba8b-97d6ece1b61a.png)
Product with min/group of rules|![Simulator Screenshot - iPhone 14 Pro - 2023-04-27 at 16 55 17](https://user-images.githubusercontent.com/8658164/234918734-4ce318b8-736b-4603-9f70-123534e2d8cc.png)|![Simulator Screenshot - iPhone 14 Pro - 2023-04-27 at 16 31 49](https://user-images.githubusercontent.com/8658164/234918792-9bd75621-0384-4dcd-a3f0-f2f7c466a10c.png)
Product without quantity rules (no change)|![Simulator Screenshot - iPhone 14 Pro - 2023-04-27 at 16 55 42](https://user-images.githubusercontent.com/8658164/234918719-9703ed29-7c8a-4e53-8850-1de7c17da6fb.png)|![Simulator Screenshot - iPhone 14 Pro - 2023-04-27 at 16 55 04](https://user-images.githubusercontent.com/8658164/234918762-63e4813c-012e-4d7f-86c5-0b9b71ef5f63.png)

---
